<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AccessControlRuleEditDialog</class>
 <widget class="QDialog" name="AccessControlRuleEditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>975</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit access control rule</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>General</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="1">
       <widget class="QLineEdit" name="ruleNameLineEdit">
        <property name="placeholderText">
         <string>add a short name for the rule here</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Rule description:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Rule name:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="ruleDescriptionLineEdit">
        <property name="placeholderText">
         <string>add a description for the rule here</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Conditions</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QComboBox" name="entityComboBox"/>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="1">
         <widget class="QComboBox" name="computerPoolsComboBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="groupsComboBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="commonComputerPoolComboBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="commonGroupComboBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QCheckBox" name="hasCommonComputerPoolsCheckBox">
          <property name="text">
           <string>is member of the same computer pool(s) as</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QCheckBox" name="hasCommonGroupsCheckBox">
          <property name="text">
           <string>has one or more groups in common with</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="isMemberOfComputerPoolCheckBox">
          <property name="text">
           <string>is member of computer pool</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QCheckBox" name="isMemberOfGroupCheckBox">
          <property name="text">
           <string>is member of group</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="QLabel" name="label">
        <property name="font">
         <font>
          <italic>true</italic>
         </font>
        </property>
        <property name="text">
         <string>If more than one condition is activated each condition has to meet in order to make the rule apply (logical AND). If only one of multiple conditions has to meet (logical OR) please create multiple access control rules.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Action</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QRadioButton" name="actionAllowRadioButton">
        <property name="text">
         <string>Allow access</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="actionDenyRadioButton">
        <property name="text">
         <string>Deny access</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="actionNoneRadioButton">
        <property name="text">
         <string>None (rule disabled)</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">actionButtonGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>ruleNameLineEdit</tabstop>
  <tabstop>ruleDescriptionLineEdit</tabstop>
  <tabstop>entityComboBox</tabstop>
  <tabstop>isMemberOfGroupCheckBox</tabstop>
  <tabstop>isMemberOfComputerPoolCheckBox</tabstop>
  <tabstop>hasCommonGroupsCheckBox</tabstop>
  <tabstop>hasCommonComputerPoolsCheckBox</tabstop>
  <tabstop>groupsComboBox</tabstop>
  <tabstop>computerPoolsComboBox</tabstop>
  <tabstop>commonGroupComboBox</tabstop>
  <tabstop>commonComputerPoolComboBox</tabstop>
  <tabstop>actionAllowRadioButton</tabstop>
  <tabstop>actionDenyRadioButton</tabstop>
  <tabstop>actionNoneRadioButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>AccessControlRuleEditDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AccessControlRuleEditDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isMemberOfGroupCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupsComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>876</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isMemberOfComputerPoolCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>computerPoolsComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>628</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>891</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasCommonGroupsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>commonGroupComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>894</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasCommonComputerPoolsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>commonComputerPoolComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>723</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>867</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="actionButtonGroup"/>
 </buttongroups>
</ui>
